source/screens/featureModalScreens/ServiceSelection.tsx

```diff
@@ -1,61 +1,92 @@
-import React, { useMemo } from "react";
-import { SafeAreaView, ScrollView } from "react-native";
+import React, { useEffect, useState } from "react";
+import {
+  ActivityIndicator,
+  Platform,
+  SafeAreaView,
+  ScrollView,
+} from "react-native";
+import styled from "styled-components/native";
+import { Text } from "../../components/atoms";
+import { BookablesService } from "../../services";
+import theme from "../../styles/theme";
 
 import ButtonList from "../../components/organisms/ButtonList";
 
+const ErrorText = styled(Text)`
+  margin: 16px;
+`;
+
 interface Props {
   onNavigate: (newRoute: string) => void;
 }
+
+type ButtonItem = {
+  buttonText: string;
+  icon: string;
+  onClick: () => void;
+};
+
 const ServiceSelection = ({ onNavigate }: Props): JSX.Element => {
-  const buttons = useMemo(
-    () => [
-      {
-        buttonText: "Mina kontakter",
-        icon: "person",
-        onClick: () => onNavigate("App"),
-        colorSchema: "red",
-        variant: "contained",
-      },
-      {
-        buttonText: "Budget- och skuldrådgivning",
-        icon: "photo-camera",
-        onClick: () => onNavigate("App"),
-      },
-      {
-        buttonText: "Kostnadsfri och oberoende energi- och klimatrådgivning",
-        icon: "photo-camera",
-        onClick: () => onNavigate("App"),
-      },
-      {
-        buttonText: "Konsumentrådgivning",
-        icon: "photo-camera",
-        onClick: () => onNavigate("App"),
-      },
-      {
-        buttonText: "Rådgivare för personer med funktionsnedsättning",
-        icon: "photo-camera",
-        onClick: () => onNavigate("App"),
-      },
-      {
-        buttonText: "Seniorrådgivning",
-        icon: "photo-camera",
-        onClick: () => onNavigate("App"),
-      },
-    ],
-    [onNavigate]
-  );
+  const [isLoading, setLoading] = useState(true);
+  const [buttons, setButtons] = useState<ButtonItem[]>([]);
+  const [error, setError] = useState<Error | undefined>(undefined);
+
+  const fetchData = async () => {
+    let canceled = false;
+    try {
+      const bookables = await BookablesService.getBookables();
+      if (!canceled) {
+        if (bookables.length === 0) {
+          throw new Error("Det finns inga tjänster tillgängliga just nu.");
```

Comment:
Im wondering if we really should throw here instead of returning the empty array. If we have an empty array it could be easier to show an error message in the UI if no services are found. And if we really throw because of an error then we have another way of showing an error message. 

what do you think?

Reply:
Sounds logical. Let's go with that.